llama_deploy/apiserver/deployment.py

```
@@ -0,0 +1,183 @@
+
+        This task is responsible for launching asyncio tasks for the core components and the services.
+        All the tasks are gathered before returning.
+        """
+        tasks = []
+        # Core components
+        tasks.append(asyncio.create_task(self._queue.launch_server()))
+        # the other components need the queue to run in order to start, give the queue some time to start
+        # FIXME: having to await a magic number of seconds is very brittle, we should rethink the bootstrap process
+        await asyncio.sleep(1)
+        cp_consumer_fn = await self._control_plane.register_to_message_queue()
+        tasks.append(asyncio.create_task(self._control_plane.launch_server()))
+        tasks.append(asyncio.create_task(cp_consumer_fn()))
+
+        # Services
+        for wfs in self._workflow_services:
+            service_task = asyncio.create_task(wfs.launch_server())
+            tasks.append(service_task)
+            consumer_fn = await wfs.register_to_message_queue()
+            control_plane_url = (
+                f"http://{self._control_plane.host}:{self._control_plane.port}"
+            )
+            await wfs.register_to_control_plane(control_plane_url)
+            consumer_task = asyncio.create_task(consumer_fn())
+            tasks.append(consumer_task)
+
+        # Run allthethings
+        await asyncio.gather(*tasks)
+
+    def _load_services(self, config: Config) -> list[WorkflowService]:
+        """Creates WorkflowService instances according to the configuration object"""
+        workflow_services = []
+        for service_id, service_config in config.services.items():
+            source = service_config.source
+            if source is None:
+                # this is a default service, skip for now
+                # TODO: check the service name is valid and supported
+                # TODO: possibly start the default service if not running already
+                continue
```

Should maybe print a warning or raise an error here if the service is unknown?

---

I plan to iterate fast here (I already have the next step ready in a branch), I'm trading unstable code for smaller PRs to ease the review process.